src/main/java/mods/railcraft/common/util/inventory/InvTools.java

```diff
@@ -78,6 +78,34 @@ public static ItemStack emptyStack() {
         return null;
     }
 
+    /**
+     * This method changes the original stack!
+     */
+    @Nullable
+    public static ItemStack decSize(@Nullable ItemStack stack) {
```

Comment:
For deplete one, does it mutate the stack itself? I don't understand it very well so I rewrote a size decrease function that always changes the stack in the argument.